src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerFileReference.cs

```diff
@@ -53,19 +55,58 @@ public sealed class AnalyzerFileReference : AnalyzerReference, IEquatable<Analyz
         /// <param name="assemblyLoader">Loader for obtaining the <see cref="Assembly"/> from the <paramref name="fullPath"/></param>
         public AnalyzerFileReference(string fullPath, IAnalyzerAssemblyLoader assemblyLoader)
         {
-            _fullPath = fullPath ?? throw new ArgumentNullException(nameof(fullPath));
+            CompilerPathUtilities.RequireAbsolutePath(fullPath, nameof(fullPath));
```

Comment:
Can this break existing hosts that use non-absolute path here, but have baked in special knowledge in the provided assembly loader to correctly load the assembly?

Reply:
Yes. But that scenario does not work anyways because we don't use assembly loader for reading the Id/DisplayName.

Reply:
Sounds good.